src/api/procedures/setAssetRequirements.ts

```diff
@@ -74,6 +75,29 @@ export async function prepareSetAssetRequirements(
     });
   }
 
+  const identityErr: string[] = [];
+
+  await P.each(flattenDeep<Condition>(requirements), async requirement => {
+    if (requirement.type === ConditionType.IsIdentity) {
+      let { identity } = requirement;
+      identity = identity instanceof Identity ? identity : new Identity({ did: identity }, context);
+      const exits = await identity.exists();
+      if (!exits) {
+        identityErr.push(identity.did);
+      }
+    }
+  });
+
+  if (identityErr.length) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'Some identities no longer exists',
```

Comment:
`Some of the passed "isIdentity" conditions reference Identities that do not exist`

Identities can't be deleted, so "no longer" doesn't apply here